main.py

```diff
@@ -0,0 +1,44 @@
+from transformers import BeitFeatureExtractor, BeitForImageClassification
+import streamlit as st
+from PIL import Image
+
+
+def predict_classes(image):
+    feature_extractor = BeitFeatureExtractor.from_pretrained('microsoft/beit-base-patch16-224-pt22k-ft22k')
+    model = BeitForImageClassification.from_pretrained('microsoft/beit-base-patch16-224-pt22k-ft22k')
+    inputs = feature_extractor(images=image, return_tensors="pt")
+    outputs = model(**inputs)
+    logits = outputs.logits
+    predicted_class_idx = logits.argmax(-1).item()
+    predicted_text = model.config.id2label[predicted_class_idx]
+    return st.text(predicted_text)
```

Comment:
Don't mix prediction and interface logic in one place. Let this function return plain text and render it together with other streamlit logic.